imap_processing/codice/codice_l1a.py

```diff
@@ -79,143 +84,143 @@ def __init__(self, table_id: int, plan_id: int, plan_step: int, view_id: int):
-                        self.num_spin_sectors,
-                        self.num_energy_steps,
-                    )
-                )
-                dims = ["epoch", "inst_az", "spin_sector", "esa_step"]
-            elif self.instrument == "hi":
-                variable_data_arr = np.array(variable_data).reshape(
-                    (
-                        1,
-                        self.num_energy_steps,
-                        self.num_positions,
-                        self.num_spin_sectors,
-                    )
-                )
-                dims = ["epoch", "esa_step", "inst_az", "spin_sector"]
+            # Reshape to 4 dimensions to allow for epoch dimension
+            reshaped_variable_data = np.expand_dims(variable_data, axis=0)
 
             # Get the CDF attributes
             cdf_attrs_key = (
                 f"{self.dataset_name.split('imap_codice_l1a_')[-1]}-{variable_name}"
             )
-            attrs = cdf_attrs.get_variable_attributes(cdf_attrs_key)
+            attrs = self.cdf_attrs.get_variable_attributes(cdf_attrs_key)
 
             # Create the CDF data variable
             dataset[variable_name] = xr.DataArray(
-                variable_data_arr,
+                reshaped_variable_data,
                 name=variable_name,
-                dims=dims,
+                dims=self.dims,
                 attrs=attrs,
             )
 
-        # Add ESA Sweep Values and acquisition times (lo only)
-        if self.instrument == "lo":
-            self.get_energy_table()
-            self.get_acquisition_times()
-            dataset["esa_sweep_values"] = xr.DataArray(
-                self.energy_table,
-                dims=["esa_step"],
-                attrs=cdf_attrs.get_variable_attributes("energy_table"),
-            )
-            dataset["acquisition_times"] = xr.DataArray(
-                self.acquisition_times,
-                dims=["esa_step"],
-                attrs=cdf_attrs.get_variable_attributes("acquisition_time_per_step"),
+        # Add support data variables based on data product
+        dataset = self.define_support_variables(dataset)
+
+        return dataset
+
+    def define_support_variables(self, dataset: xr.Dataset) -> xr.Dataset:
+        """
+        Define and add 'support' CDF data variables to the dataset.
+
+        These variables include instrument metadata, energies, times, etc. that
+        help further define the L1a CDF data product. The variables included
+        depend on the data product being made.
+
+        Parameters
+        ----------
+        dataset : xarray.Dataset
+            ``xarray`` dataset for the data product.
+
+        Returns
+        -------
+        dataset : xarray.Dataset
+            ``xarray`` dataset for the data product, with added support variables.
+        """
+        for variable_name in self.support_variables:
+            if variable_name == "energy_table":
+                variable_data = self.get_energy_table()
```

Comment:
Is this function getting energy value in engineering units or is this getting energy step value, Eg. [0,1,2,....127]?

Reply:
It is getting the ESA sweep values technically, which I am calling the "energy table". This is what it is ultimately reading in: https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/dev/imap_processing/codice/data/esa_sweep_values.csv